components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/ComponentRegistryResolverImpl.kt

```diff
@@ -69,127 +70,74 @@ class ComponentRegistryResolverImpl(
-        return with(getJiraComponentVersion(component, version)) {
-            val buildVersion = this.component.componentVersionFormat.buildVersionFormat.formatVersion(
-                numericVersionFactory,
-                this.version
-            )
-            getJiraComponentVersionRange(component, buildVersion)
-                .let {
-                    ModelConfigPostProcessor(
-                        ComponentVersion.create(
-                            component,
-                            buildVersion
-                        ), versionNames
-                    ).resolveVariables(it.vcsSettings)
-                }
-        }
+        val (jiraComponentVersion, jiraComponentVersionRange) =
+            getJiraComponentVersionToRangeByComponentAndVersion(component, version)
+        val buildVersion = jiraComponentVersion.component.componentVersionFormat.buildVersionFormat.formatVersion(
+            numericVersionFactory, jiraComponentVersion.version
+        )
+        return ModelConfigPostProcessor(ComponentVersion.create(component, buildVersion), versionNames)
+            .resolveVariables(jiraComponentVersionRange.vcsSettings)
     }
 
-    override fun getJiraComponentByProjectAndVersion(projectKey: String, version: String): JiraComponentVersion {
-        val projectKeyToJiraComponentVersionRangeMap =
-            jiraParametersResolver.componentConfig.projectKeyToJiraComponentVersionRangeMap
-        return getJiraComponentVersion(projectKey, version, projectKeyToJiraComponentVersionRangeMap)
-    }
+    override fun getJiraComponentByProjectAndVersion(projectKey: String, version: String) =
+        getJiraComponentVersionToRangeByProjectAndVersion(projectKey, version).first
 
-    override fun getJiraComponentsByProject(projectKey: String): Set<String> {
-        return getProjectJiraComponentVersionRanges(projectKey)
-            .map { it.componentName }
-            .toSet()
-    }
+    override fun getJiraComponentsByProject(projectKey: String) =
+        getJiraComponentVersionRangesByProject(projectKey).map { it.componentName }.toSet()
 
-    override fun getJiraComponentVersionRangesByProject(projectKey: String): Set<JiraComponentVersionRange> {
-        return getProjectJiraComponentVersionRanges(projectKey)
-            .toSet()
-    }
+    override fun getJiraComponentVersionRangesByProject(projectKey: String) =
+        jiraParametersResolver.componentConfig.projectKeyToJiraComponentVersionRangeMap[projectKey]?.toSet()
+            ?: throw NotFoundException("Project '$projectKey' is not found")
 
     override fun getComponentsDistributionByJiraProject(projectKey: String): Map<String, Distribution> {
-        return getProjectJiraComponentVersionRanges(projectKey)
+        return getJiraComponentVersionRangesByProject(projectKey)
             .groupBy { it.componentName }
-            .mapValues {
+            .mapValues { entry ->
```

Comment:
may be use destruction like (_, value) -> ?

Reply:
I've just fixed hiding of `it` here. But destruction is better indeed. Fixed.